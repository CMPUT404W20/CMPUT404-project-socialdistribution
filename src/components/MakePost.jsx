import React, { Component } from "react";
import "../styles/makePost.scss";
import SendIcon from "@material-ui/icons/Send";
import ImageOutlinedIcon from "@material-ui/icons/ImageOutlined";
import Modal from "react-bootstrap/Modal";
import icon from "../images/markdown-icon.svg";

class MakePost extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modalShow: false,
      post: "",
    };
  }

  handleTextChange = (event) => {
    this.setState({ post: event.target.value });
  };

  handleSubmit = () => {
    const { post } = this.state;
    // eslint-disable-next-line no-alert
    alert(post);
  };

  renderModal = () => {
    this.setState({ modalShow: true });
  };

  Modal = () => {
    const { modalShow } = this.state;
    const handleClose = () => this.setState({ modalShow: false });
    return (
      <div className="modal-upload">
        <Modal show={modalShow} onHide={handleClose} animation={false}>
          <Modal.Header closeButton>
            <Modal.Title>Upload images</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <input
              accept="image/*"
              className="upload-image"
              id="contained-button-file"
              multiple
              type="file"
            />
          </Modal.Body>
          <Modal.Footer className="upload-button-wrapper">
            <button
              type="button"
              className="upload-button"
              onClick={handleClose}
            >
              Upload
            </button>
          </Modal.Footer>
        </Modal>
      </div>
    );
  };

  render() {
    return (
      <div className="makePost">
        <div className="block">
          <div className="form-row-1">
            <b>NEW POST</b>
            <select className="privacy-select">
              <option selected value="public">
                Anyone
              </option>
              <option value="another author">Specific author</option>
              <option value="friends">Friends</option>
              <option value="mutual friends">Mutual friends</option>
              <option value="local friends">Local friends</option>
              <option value="private">Private</option>
            </select>
          </div>
          {this.Modal}
          <form className="row-2-input" action="submit">
            <textarea
              placeholder="What's on your mind?"
              onChange={this.handleTextChange}
            />
            <div className="row-3-buttons">
              <img
                className="markdown-icon icon"
                src={icon}
                width="20pt"
                alt="markdown"
              />
              <ImageOutlinedIcon
                className="image-icon icon"
                onClick={this.renderModal}
              />
              <button
                type="submit"
                className="post-button icon"
                onClick={this.handleSubmit}
              >
                <span>POST</span>
                <SendIcon className="post-icon" />
              </button>
            </div>
          </form>
        </div>
        {this.Modal()}
      </div>
    );
  }
}
export default MakePost;
