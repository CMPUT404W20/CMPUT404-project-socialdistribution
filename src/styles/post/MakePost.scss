$box-width: 100%;
$box-height: 205px;
$border-color: #dadada;
$line-separator-color: #c1c1c1;
$accent-color: #4B84F2;
$icon-color: #787878;

.icon {
  cursor: pointer;
}

.makePost {
  margin-top: 90px;
  background-color: white;
  box-sizing: border-box;
  box-shadow: 0px 0px 6px rgba(202, 202, 202, 0.25);
  border-radius: 4px;
  width: $box-width;
  height: $box-height;
  display: flex;
  justify-content: center;
}

.block {
  display: flex;
  flex-direction: column;
  width: 90%;
}

.form-row-1 {
  display: flexbox;
  flex-direction: row;
  justify-content: center;
  border-bottom: 1px solid $line-separator-color;
  height: fit-content;
  padding-top: 10pt;
  padding-bottom: 3pt;

  & > b {
    font-style: normal;
    font-weight: 500;
    font-size: 10pt;
    text-transform: uppercase;
    color: #181818;
  }
}

select {
  font-style: normal;
  font-size: 10pt;
  margin-left: 5%;
}

.row-2-input {
  height: 40%;
  margin-top: 1.5rem;
}

.row-2-input > textarea {
  width: 100%;
  font-size: 12pt;
  height: 100%;
  border: none;
  outline: 0;
}

.post-button,
.upload-button {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border: none;
  background-color: $accent-color;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: 2px;
  height: 1.8rem;
  color: white;
  padding-left: 1.8rem;
  padding-right: 0.9rem;
  font-style: normal;
  font-weight: 500;
  font-size: 10pt;
  text-align: center;
  outline: 0;

  > .post-icon {
    color: white;
    -webkit-transform: rotate(-40deg);
    -moz-transform: rotate(-40deg);
    -ms-transform: rotate(-40deg);
    -o-transform: rotate(-40deg);
    transform: rotate(-40deg);
    margin-bottom: 0.3rem;
    width: 2rem;
    height: 1.3rem;
  }
}

.row-3-buttons {
  padding-top: 0.5rem;
  display: flex;
  float: right;
  justify-content: right;
  justify-content: space-between;
  flex-direction: row;
  align-items: flex-end;
  width: 11.5rem;
  // The icon has some space at the bottom by default and is
  // not aligned to the bottom line without changing the margin
  > .image-icon {
    color: $icon-color;
    margin-bottom: -0.18rem;
  }
}

.upload-button-wrapper {
  display: flex;
  width: 100%;
  justify-content: center;
  
  > .upload-button {
    width: 100%;
  }
}
