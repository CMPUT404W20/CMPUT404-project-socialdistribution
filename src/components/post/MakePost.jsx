/* eslint-disable max-classes-per-file */
import React, { Component } from "react";
import PropTypes from "prop-types";
import "../../styles/post/MakePost.scss";
import SendIcon from "@material-ui/icons/Send";
import ImageOutlinedIcon from "@material-ui/icons/ImageOutlined";
import Modal from "react-bootstrap/Modal";

function UploadImageModal(props) {
  const { onHide, show } = props;

  return (
    <Modal onHide={onHide} show={show} className="upload-image-modal">
      <Modal.Header closeButton>
        <Modal.Title>Upload images</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <input
          accept="image/*"
          className="upload-image"
          id="contained-button-file"
          multiple
          type="file"
        />
      </Modal.Body>
      <Modal.Footer className="upload-button-wrapper">
        <button
          type="button"
          className="upload-button"
          onClick={onHide}
        >
          Upload
        </button>
      </Modal.Footer>
    </Modal>
  );
}

UploadImageModal.propTypes = {
  show: PropTypes.func.isRequired,
  onHide: PropTypes.func.isRequired,
};


class MakePost extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modalShow: false,
      post: "",
    };
  }

  handleTextChange = (event) => {
    this.setState({ post: event.target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();

    const { post } = this.state;
    // eslint-disable-next-line no-alert
    alert(post);
  };

  renderModal = () => {
    this.setState({ modalShow: true });
  };

  render() {
    const { modalShow } = this.state;
    const handleClose = () => this.setState({ modalShow: false });
    return (
      <div className="make-post-wrapper">
        <div className="make-post-content">
          <div className="make-post-header">
            <b>NEW POST</b>
            <select className="privacy-select">
              <option selected value="public">
                Anyone
              </option>
              <option value="another author">Specific author</option>
              <option value="friends">Friends</option>
              <option value="mutual friends">Mutual friends</option>
              <option value="local friends">Local friends</option>
              <option value="private">Private</option>
            </select>
          </div>
          <UploadImageModal show={modalShow} onHide={handleClose} />
          <form className="make-post-input-wrapper" action="submit">
            <textarea
              placeholder="What's on your mind?"
              className="post-text-area"
              onChange={this.handleTextChange}
            />
            <div className="make-post-buttons-wrapper">
              <ImageOutlinedIcon
                className="upload-image-icon icon"
                onClick={this.renderModal}
              />
              <button
                type="submit"
                className="post-button icon"
                onClick={this.handleSubmit}
              >
                <span>POST</span>
                <SendIcon className="post-icon" />
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
export default MakePost;
