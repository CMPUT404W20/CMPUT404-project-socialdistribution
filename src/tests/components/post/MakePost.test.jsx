import React from "react";
import { shallow } from "enzyme";
import TextareaAutosize from "react-textarea-autosize";
import MakePost from "../../../components/post/MakePost";

describe("Post box Component", () => {
  it("should render correctly", () => {
    const component = shallow(<MakePost />);
    expect(component).toMatchSnapshot();
  });

  it("should have a text area", () => {
    const component = shallow(<MakePost />);
    expect(component.find(TextareaAutosize)).toHaveLength(1);
  });

  it("should show the modal after clicking on image icon", () => {
    const component = shallow(<MakePost />);
    const { uploadModalVisibility } = component.state();
    expect(uploadModalVisibility).toBe(false);
    setTimeout(() => {
      component.find(".image-icon").simulate("click");

      jest.runOnlyPendingTimers();
      component.update();
      expect(uploadModalVisibility).toBe(true);
    }, 500);
  });
});
