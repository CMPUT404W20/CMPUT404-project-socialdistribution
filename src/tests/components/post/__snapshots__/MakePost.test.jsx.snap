// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post box Component should render correctly 1`] = `
<div
  className="make-post-wrapper"
>
  <div
    className="make-post-content"
  >
    <div
      className="make-post-header"
    >
      <b>
        NEW POST
      </b>
      <select
        className="privacy-select"
        defaultValue="PUBLIC"
        onChange={[Function]}
      >
        <option
          value="PUBLIC"
        >
          Anyone
        </option>
        <option
          value="PRIVATE"
        >
          Specific author
        </option>
        <option
          value="FRIENDS"
        >
          Friends
        </option>
        <option
          value="FOAF"
        >
          Mutual friends
        </option>
        <option
          value="SERVERONLY"
        >
          Server Only
        </option>
        <option
          value="UNLISTED"
        >
          Private
        </option>
      </select>
    </div>
    <UploadImageModal
      onHide={[Function]}
      onUpload={[Function]}
      show={false}
    />
    <PostPreviewModal
      imageObjectUrl=""
      onHide={[Function]}
      postContent=""
      show={false}
    />
    <form
      action="submit"
      className="make-post-input-wrapper"
    >
      <TextareaAutosize
        className="post-text-area"
        inputRef={[Function]}
        onChange={[Function]}
        onHeightChange={[Function]}
        placeholder="What's on your mind?"
        useCacheForDOMMeasurements={false}
        value=""
      />
      <div
        className="make-post-buttons-wrapper"
      >
        <ImageOutlinedIcon
          className="icon"
          onClick={[Function]}
        />
        <button
          className="post-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Post
        </button>
      </div>
    </form>
  </div>
</div>
`;
